app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala

```diff
@@ -194,6 +243,87 @@ class GithubConnector @Inject()(
     }
 }
 
+object GithubConnector {
+
+  final case class GraphqlQuery(
+    query: String,
+    variables: Map[String, JsValue] = Map.empty
+  ) {
+
+    def withVariable(name: String, value: JsValue): GraphqlQuery =
+      copy(variables = variables + (name -> value))
```

Comment:
what happens if not all the variables are bound? We get a `400` from github?
I guess the tests confirm we always bind all variable.

Reply:
Yep, would get a 400 (if the variable is required that is, marked with `!`.)

But you've made me think about the case of a team not existing—pretty sure we'll get a 200 there with a GQL error object, so that doesn't fit in with the freebies we'd get if it returned 404. Will update.